test/spec/proxySpec.js

```diff
@@ -1146,56 +1142,7 @@ describe('proxy', function() {
       // }, 100);
     });
   });
-  describe('Already proxified values', function() {
-    it("shouldn't re-proxify (it should deep clone) proxified values from different instances", function() {
-
-      var obj = {
-        one: [1, 2, 3, 4, 5],
-        two: [6, 7, 6, 5, 4]
-      };
-
-      var countOne = 0;
-      var observedObjOne = new JSONPatcherProxy(obj).observe(true, function() {
-        countOne++;
-      });
-
-      var countTwo = 0;
-      var observedObjTwo = new JSONPatcherProxy(obj).observe(true, function() {
-        countTwo++;
-      });
-
-      //control test, to make sure tests are correct
-      expect(countOne).toReallyEqual(0);
-      observedObjOne.one[0] = 100;
-      expect(countOne).toReallyEqual(1);
-
-      // isolate the proxified array
-      var observedArray = observedObjOne.one;
-
-      // change the proxified array
-      observedArray[1] = 200;
-
-      // expect a patch
-      expect(countOne).toReallyEqual(2);
-
-      // make sure second observer never went off
-      expect(countTwo).toReallyEqual(0);
-
-      // add an already proxified array 
-      observedObjTwo.three = observedArray;
-
-      // expect a patch from second observer (add /three)
-      expect(countTwo).toReallyEqual(1);
-
-      //change the array in second observer
-      observedObjTwo.three[2] = 100;
-
-      // expect a patch from second observer (replace /three/2/)
-      expect(countTwo).toReallyEqual(2);
-
-      // first observer should NOT emit a patch
-      expect(countOne).toReallyEqual(2);      
-    });
```

Comment:
Have we changed the behavior? are we proxify already proxified objects?

Reply:
Yup. I removed deep cloning from here and from initial proxification for better performance.

Reply:
How would the new version behave if you cross proxify the object?

Reply:
Both instances would emit patches at once. As one would expect.

Reply:
Cool.

How about edge cases, like:
```js
var obj = {
    one: [1, 2, 3, 4, 5],
    two: {
        raz: [6, 7, 6, 5, 4]
    }
};
var observedObj = new JSONPatcherProxy(obj).observe(true, callback);
var observedObjTwo = new JSONPatcherProxy(observedObj.two).observe(true, callback2);

delete observedObj.two.raz;
```
or
```js
var observedObj = new JSONPatcherProxy(obj).observe(true, callback);
var observedObjTwo = new JSONPatcherProxy(obj.two).observe(true, callback2);

delete observedObj.two.raz;
```

Reply:
Good catch. This causes an issue. 

Proxifying an object actually modifies it, because it does
`obj.prop = Proxify(obj.prop)`

So the second constructor would emit patches that are unexpected. Check here. 

https://jsfiddle.net/L0myd8p1/

I'll go back to deep cloning but will find a faster alternative.

Reply:
maybe we can separate this issue as it's not a natural case. So, we will release fixed performance first.

Reply:
Good point. We shouldn't cater for other people on our own expense. I guess a boolean in the constructor would do.